src/StoreApi/Utilities/CartController.php

```diff
@@ -803,9 +806,37 @@ function( $key, $package, $index ) {
 			);
 		}
 
-		return $calculate_rates ? wc()->shipping()->calculate_shipping( $packages ) : $packages;
+		$packages = $calculate_rates ? wc()->shipping()->calculate_shipping( $packages ) : $packages;
+
+		remove_filter( 'woocommerce_get_shipping_methods', [ $this, 'enable_local_pickup_without_address' ] );
+
+		return $packages;
 	}
 
+	/**
+	 * We want to make local pickup always avaiable without checking for a shipping zone or address.
+	 *
+	 * @param array $shipping_methods Package we're checking against right now.
+	 * @return array $shipping_methods Shipping methods with local pickup.
+	 */
+	public function enable_local_pickup_without_address( $shipping_methods ) {
+		$shipping_zones = \WC_Shipping_Zones::get_zones( 'admin' );
+		$worldwide_zone = new \WC_Shipping_Zone( 0 );
+		$all_methods    = array_map(
+			function( $_shipping_zone ) {
+				return $_shipping_zone['shipping_methods'];
+			},
+			$shipping_zones
+		);
+		$all_methods    = array_merge_recursive( $worldwide_zone->get_shipping_methods( false, 'admin' ), ...$all_methods );
+		$local_pickups  = array_filter(
+			$all_methods,
+			function( $method ) {
+				return 'local_pickup' === $method->id;
+			}
+		);
+		return array_merge( $shipping_methods, $local_pickups );
```

Comment:
And we merge it with the returned shipping zones.

Now WooCommerce Cart would operate on this list.